lib/elixir/test/elixir/file_test.exs

```diff
@@ -1322,6 +1322,26 @@ defmodule FileTest do
     assert Enum.count(stream) == 2
   end
 
+  test "stream keeps BOM" do
+    src = fixture_path("utf8_bom.txt")
+    bom_line = src
+    |> File.stream!()
+    |> Enum.take(1)
+    |> List.first()
```

Comment:
Since we know the shape of this, I think the right way to do this is either use `hd/1` in place of `List.first/1`, or even do

```elixir
[bom_line] =
  src
  |> File.stream!()
  |> Enum.take(1)
```

Wdyt?